index.html

```diff
@@ -3102,22 +3116,44 @@ <h3>Verify Proof Sets and Chains</h3>
 Set <var>allProofs</var> to </var><var>securedDocument</var>.<var>proof</var>.
           </li>
           <li>
-For each <var>proof</var> in <var>allProofs</var>, run steps 2 through 11 of the algorithm in
+For each <var>proof</var> in <var>allProofs</var>, do the following steps:
+          </li>
+          <ol class="algorithm">
+            <li>
+Let <var>matchingProofs</var> be an empty list.
+            </li>
+            <li>
+If <var>proof</var> contains a <code>previousProof</code> attribute and that
+attribute is a string, add the element from <var>allProofs</var> with an
+<code>id</code> attribute matching <code>previousProof</code> to matchingProofs.
+If a proof with <code>id</code> does not exist in <var>allProofs</var>, an error
+MUST be raised and SHOULD convey an error type of
+<a href="#MALFORMED_PROOF_ERROR">MALFORMED_PROOF_ERROR</a>.
+If the <code>previousProof</code> attribute is an array, add each element from
+<var>allProofs</var> with an <code>id</code> attribute that matches an element
+of that array. If any matches are missing, an error MUST be raised and SHOULD
+convey an error type of
+<a href="#MALFORMED_PROOF_ERROR">MALFORMED_PROOF_ERROR</a>.
+            </li>
+            <li>
+Let <var>unsecuredDocument</var> be a copy of <var>securedDocument</var> with
+the proof value removed and then set <var>unsecuredDocument.proof</var>
+to <var>matchingProofs</var>.
+            </li>
+            <li>
+Run steps 2 through 4 to validate the contents of <var>proof</var>.
+            </li>
+            <li>
+Run steps 6 through 11 of the algorithm in
 section <a href="#verify-proof"></a>; if no exceptions are raised, associate
 the <var>isProofVerified</var> value with this <var>proof</var>.
-          </li>
-          <li>
-For each <var>proof</var> in <var>allProofs</var> that contains a
-<code>previousProof</code> attribute, modify the associated
-<var>isProofVerified</var> by iteratively checking that the proof identified by
-the <code>id</code> value in <var>previousProof</var>, along with every
-<code>previousProof</code> up the chain, is valid. If <code>previousProof</code> is an
-array, then all previous proofs identified by the array elements
-MUST be valid for the current proof to be considered valid. Cycle detection,
-such as keeping a list of previously visited proofs, MUST be used to guard
-against proof cycles. If a cycle is detected in a proof chain,
-then a `PROOF_CHAIN_CYCLE_ERROR` MUST be raised.
```

Comment:
So I believe we still need some of this text. In short, each of the proofs referenced (by `id`) in `previousProof` need to be verified prior to checking the current proof. And if a cycle is detected when doing this checking, the error mentioned must be raised (I see we have text that adds a definition for it below, but without this text there would nothing that would throw it either).

So I expected we can simplify the above text considerably, but we can't remove it entirely. We can probably insert some text above (I'll try to add a suggestion or comment above for that).

Reply:
Actually, ignore this comment -- I see that a different, simpler change is possible, I'll do another review.

Reply:
@dlongley -- Could you `<strike>` the comment(s) that should be ignored? This will help save us the time of reading those comments before we get to the "ignore" flag. :-)